common/djangoapps/student/views.py

```diff
@@ -1854,6 +1856,16 @@ def _enroll_user_in_pending_courses(student):
                 )
 
 
+def _record_registration_attribution(request, user):
+    """
+    Attribute this user's registration to the referring affiliate, if
+    applicable.
+    """
+    affiliate_id = request.COOKIES.get(settings.AFFILIATE_COOKIE_NAME)
+    if user is not None and affiliate_id is not None:
```

Comment:
I prefer to do the explicit `is None` test, because it can't be overridden and weak typing is sneaky sometimes.